### PR TITLE
CORE-1732: updated the DE UI to use the quota values from the resource usage endpoint

### PR DESCRIPTION
**Pull request by slr71**

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -98,7 +98,9 @@ export default {
         INSTANT_LAUNCH_LISTING: "listing",
     },
     VICE_LOGS_POLLING_INTERVAL: 10000,
-    DATA_STORAGE_QUOTA_LIMIT: 2 ** 30 * 100, // 100 GiB
+    DATA_STORAGE_QUOTA_LIMIT_DEFAULT: 50 * 2 ** 30,
```
Comment (ianmcorvidae) [inline src/constants.js:101]: Why the change to 50? I thought the default was still 100, but I probably missed something, heh.

File: src/constants.js [hunk 2]
```diff
@@ -98,7 +98,9 @@ export default {
...
+    DATA_STORAGE_RESURCE_NAME: "data.size",
```
Comment (ianmcorvidae) [inline src/constants.js:102]: Looks like a typo in `resource` here

File: src/components/dashboard/dashboardItem/util.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * @returns {number} the quota value to use.
+ */
+export const getUserQuota = (resourceTypeName, userPlan) => {
+    const quotas = userPlan?.quotas;
+    const defaultQuotaValue = defaultQuotaFor[resourceTypeName] || 0;
+
+    // Search for the quota with the given resource type name.
+    if (quotas) {
+        for (const quota of quotas) {
+            console.log(quota);
```
Comment (psarando) [inline src/components/dashboard/dashboardItem/util.js:34]: Looks like debug logging that can be removed now.

**Review by ianmcorvidae — APPROVED** [review]
Seems reasonable to me (other than the small typo).

File: src/constants.js (hunk 2 above)
Comment (slr71) [reply src/constants.js:102]: D'oh! Good catch. I'll fix it 🤣

File: src/constants.js (hunk 1 above)
Comment (slr71) [reply src/constants.js:101]: The limit associated with the basic plan was changed to 50 GiB.

File: stories/dashboard/CPUConsumption.stories.js [hunk 4]
```diff
@@ -5,18 +5,182 @@ export default {
...
-                user_id: "626717be-575e-11ea-8038-008cfa5ae621",
-                username: "sonora_test@iplantcollaborative.org",
-                total: "168.4269529152434",
-                effective_start: "2022-02-03T00:34:41.88739Z",
-                effective_end: "2023-02-03T00:34:41.88739Z",
-                last_modified: "2022-02-03T02:38:17.906799Z",
-            }}
-        />
-    );
+const CPUConsumptionTemplate = (args) => {
```
Comment (psarando) [inline stories/dashboard/CPUConsumption.stories.js:8]: 👍

**Review by psarando — APPROVED** [review]
LGTM 👍 ✨

File: src/components/dashboard/dashboardItem/util.js (hunk 3 above)
Comment (slr71) [reply src/components/dashboard/dashboardItem/util.js:34]: Thanks for catching that; I meant to remove it but forgot to

**Comment by slr71** [comment]
Thanks for the reviews!
